llvm/include/llvm/Support/GenericDomTree.h

```diff
@@ -882,9 +879,9 @@ class DominatorTreeBase {
     // changed.
     if (!NewBBIDom) return;
 
-    for (i = i + 1; i < PredBlocks.size(); ++i) {
-      if (isReachableFromEntry(PredBlocks[i]))
-        NewBBIDom = findNearestCommonDominator(NewBBIDom, PredBlocks[i]);
+    for (const auto &Pred : llvm::drop_begin(PredBlocks)) {
```

Comment:
That's not right -- this loop is supposed to pick up where the last one stopped, not skip just the first element.

The code is still correct because the NCD operation is idempotent, but it makes the code less efficient.

Reply:
Thank you for pointing this out!  I've revised the patch.  Please take a look.